cairo/src/context.rs

```diff
@@ -474,17 +479,17 @@ impl Context {
     }
 
     #[doc(alias = "cairo_mask")]
-    pub fn mask(&self, pattern: &Pattern) -> Result<(), Error> {
-        pattern.status()?;
-        unsafe { ffi::cairo_mask(self.0.as_ptr(), pattern.to_raw_none()) };
+    pub fn mask(&self, pattern: impl AsRef<Pattern>) -> Result<(), Error> {
```

Comment:
```suggestion
    pub fn mask(&self, pattern: &impl AsRef<Pattern>) -> Result<(), Error> {
```

etc or not? There's no reason to take these things by value :)

Reply:
I thought, Impl AsRef<T> would already take things by ref

Reply:
Indeed: https://doc.rust-lang.org/std/convert/trait.AsRef.html#impl-AsRef%3CU%3E-for-%26T

Does that work? You can now pass `&surface` and `surface`, I guess? I hope we have some code in the examples that does this :)

Reply:
Just tried to change a test and it works with both `&surface` / `surface`